src/apiMocks/mockData.ts

```diff
@@ -98,7 +98,14 @@ export function createRandomBreach(
     : BreachDataTypes;
   const dataClasses =
     options.dataClasses ??
-    faker.helpers.arrayElements(Object.values(dataClassTypes));
+    (Array.isArray(options.dataClassesEffected)
+      ? options.dataClassesEffected
+          .map(
+            (affectedObj) =>
+              Object.keys(affectedObj) as SubscriberBreach["dataClasses"],
+          )
+          .flat()
+      : faker.helpers.arrayElements(Object.values(dataClassTypes)));
```

Comment:
This is done so that the otherwise randomly generated `dataClasses` of the breach are actually containing the values of `dataClassesEffected`. Otherwise, this would lead to issues since a user could not be affected by a data class that the breach does not include.

Reply:
Ah yes, but you're right that that's not obvious from the code itself. I wanted to add a comment to explain it, but to be honest I'm still not entirely sure why a `SusbcriberBreach` has both `dataClasses` and `dataClassesEffected` - added an attempt to describe what's happening in 8000ff487 anyway.